trio/_core/_io_windows.py

```diff
@@ -109,6 +110,65 @@ def _handle(obj):
         return obj
 
 
+async def WaitForSingleObject(handle):
+    """Async and cancellable variant of kernel32.WaitForSingleObject().
+
+    Args:
+      handle: A win32 handle.
+
+    """
+    # Quick check; we might not even need to spawn a thread. The zero
+    # means a zero timeout; this call never blocks. We also exit here
+    # if the handle is already closed for some reason.
+    retcode = kernel32.WaitForSingleObject(handle, 0)
+    if retcode != ErrorCodes.WAIT_TIMEOUT:
+        return
+
+    # :'(  avoid circular imports
+    from .._threads import run_sync_in_worker_thread
+
+    class StubLimiter:
+        def release_on_behalf_of(self, x):
+            pass
+
+        async def acquire_on_behalf_of(self, x):
```

Comment:
Since nothing here depends on access to the internals of the `WindowsIOManager`, I think we can move it out into some place like `trio/_wait_for_single_object.py`.

Reply:
Done. This also gets rid of that circular import.